src/commands/appdistribution-distribute.ts

```diff
@@ -210,56 +224,78 @@ export const command = new Command("appdistribution:distribute <release-binary-f
     await requests.distribute(releaseName, testers, groups);
 
     // Run automated tests
-    if (testDevices?.length) {
-      utils.logBullet("starting automated tests (note: this feature is in beta)");
-      const releaseTest = await requests.createReleaseTest(
-        releaseName,
-        testDevices,
-        loginCredential,
-      );
-      utils.logSuccess(`Release test created successfully`);
+    if (testDevices.length) {
+      utils.logBullet("starting automated test (note: this feature is in beta)");
+      const releaseTestPromises: Promise<ReleaseTest>[] = [];
+      if (!testCases.length) {
+        // fallback to basic automated test
+        releaseTestPromises.push(
+          requests.createReleaseTest(releaseName, testDevices, loginCredential),
+        );
+      } else {
+        for (const testCaseId of testCases) {
+          releaseTestPromises.push(
+            requests.createReleaseTest(
+              releaseName,
+              testDevices,
+              loginCredential,
+              `${appName}/testCases/${testCaseId}`,
+            ),
+          );
+        }
+      }
+      const releaseTests = await Promise.all(releaseTestPromises);
+      utils.logSuccess(`${releaseTests.length} release test(s) started successfully`);
       if (!options.testNonBlocking) {
-        await awaitTestResults(releaseTest.name!, requests);
+        const releaseTestNames = new Set(releaseTests.map((rt) => rt.name!!));
+        await awaitTestResults(releaseTestNames, requests);
       }
     }
   });
 
 async function awaitTestResults(
-  releaseTestName: string,
+  releaseTestNames: Set<string>,
   requests: AppDistributionClient,
 ): Promise<void> {
   for (let i = 0; i < TEST_MAX_POLLING_RETRIES; i++) {
-    utils.logBullet("the automated tests results are pending");
+    utils.logBullet("the automated test results are pending");
```

Comment:
Not new code, but log lines should generally be capitalized
```suggestion
    utils.logBullet("The automated test results are pending...");
```

Reply:
Done.